src/invidious/yt_backend/youtube_api.cr

```diff
@@ -434,8 +425,27 @@ module YoutubeAPI
       )
     end
 
+    if {{ !flag?(:disable_quic) }} && CONFIG.use_quic
+      body = response.body
+    else
+      # Decompress the body ourselves, when using HTTP::Client given that
+      # auto-decompress is broken in the Crystal stdlib.
+      # Read more:
+      #  - https://github.com/iv-org/invidious/issues/2612
+      #  - https://github.com/crystal-lang/crystal/issues/11354
+      #
+      case response.headers["Content-Encoding"]?
+      when "gzip"
+        body = Compress::Gzip::Reader.new(response.body_io, sync_close: true)
```

Comment:
Multiple encoding types can be specified in `Content-Encoding`. 

> ```
> // Multiple, in the order in which they were applied
> Content-Encoding: deflate, gzip
> ```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding#syntax

Reply:
The client may send multiple items, but the reply will only be in one format! The server won't return the content duplicated, once in gzip and another one in deflate.

Reply:
No, what? Of course content isn't duplicated. The server can however return content that's encoded in two or more different formats, hence `Content-Encoding`. 

The above example is first compressed with `deflate` and then `gzip`. So in other to fetch the uncompressed content you'd need to decompress `gzip` and then decompress with `deflate`.

Reply:
> The above example is first compressed with `deflate` and then `gzip`. So in other to fetch the uncompressed content you'd need to decompress `gzip` and then decompress with `deflate`.

Is the native HTTP client from Crystal really going to do that if it encounters a double compression?

Reply:
Mozilla specifies that explicitly on their page for the `Content-Encoding` header, so I'd assume that is on the spec. Which means, stdlib HTTP::Client should most likely support that

Reply:
> The above example is first compressed with `deflate` and then `gzip`. So in other to fetch the uncompressed content you'd need to decompress `gzip` and then decompress with `deflate`.

Oooooh! I didn't know that it was possible! I'll fix that, then.

> Is the native HTTP client from Crystal really going to do that if it encounters a double compression?

Well, no, the Crystal HTTP::Client doesn't support that. I copied the code above directly from the Crystal stdlib!